libs/@local/graph/postgres-store/src/store/postgres/ontology/entity_type.rs

```
@@ -1753,40 +1753,79 @@ where
-                                entity_type_id.into(),
-                            )))),
-                            context: RequestContext::default(),
-                        },
-                        &policy_context,
-                    )
-                    .change_context(QueryError)
-                    .map(|authorized| match authorized {
-                        Authorized::Always => true,
-                        Authorized::Never => false,
-                        Authorized::Partial(partial) => {
-                            tracing::error!(
-                                "Instantiation checking is not supported for partial \
-                                 authorization:\n
+        entity_type_id_set
+            .into_iter()
+            .map(|(base, parents)| {
+                // We need to check the base entity type and all its parents
+                // to see if the user can instantiate it.
+                for entity_type_id in iter::once(base).chain(parents) {
+                    let allowed = policy_set
+                        .evaluate(
+                            &Request {
+                                actor: Some(actor),
+                                action: ActionName::Instantiate,
+                                resource: Some(&PartialResourceId::EntityType(Some(Cow::Owned(
+                                    EntityTypeId::new(entity_type_id),
+                                )))),
+                                context: RequestContext::default(),
+                            },
+                            &policy_context,
+                        )
+                        .change_context(QueryError)
+                        .map(|authorized| match authorized {
+                            Authorized::Always => true,
+                            Authorized::Never => false,
+                            Authorized::Partial(partial) => {
+                                tracing::error!(
```

shouldn't this error out instead of failing silently?

---

This is actually something which should never happen because we define an ID above. I could make it an `unreachable!` instead

---

757ac38b05dd520a21f360a684fb686abb3b2f7a